pkg/controllers/vdb/upgrade.go

```diff
@@ -154,11 +155,11 @@ func (i *UpgradeManager) isVDBImageDifferent(ctx context.Context, sandbox string
 		}
 
 		if cntImage != targetImage {
-			return true, nil
+			differentImgFromVdb = true
```

Comment:
Why do we need this change? Here if `differentImgFromVdb` is true, we should stop the for loop and return.

Reply:
Say you have 3 statefulsets  sts1: img2 sts2: img2 sts3: img1 and in vdb image=img1. Normally we should not trigger upgrade but given that the 1st stafulset of the loop (sts1) has a different image from vdb, we will. With this change, you wait until you have finished comparing the images across statefulsets.